File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Par_OrarioIntervalloHHService/Models/Par_OrarioIntervalloHHModel.cs
using nvxapp.server.data.Extensions;
using nvxapp.server.service.ClientServer_Service.ModelsBase;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_OrarioIntervalloHHService.Models
{
    public class Par_OrarioIntervalloHHModel
    {

        public int Id { get; set; }
        public int IdPar_Orario { get; set; }
        //public int IdAz_SubCommessaAttivita { get; set; }
        public TimeOnly? Dalle { get; set; }
        public TimeOnly? Dalle_Limite_SX { get; set; }
        public TimeOnly? Dalle_Limite_DX { get; set; }
        public TimeRoundInterval Dalle_Arrotondamento { get; set; }
        public RoundDirection Dalle_Arrotondamento_Verso { get; set; }
        public Boolean Dalle_Use_4_Match { get; set; }

        public TimeOnly? Alle { get; set; }
        public TimeOnly? Alle_Limite_SX { get; set; }
        public TimeOnly? Alle_Limite_DX { get; set; }
        public TimeRoundInterval Alle_Arrotondamento { get; set; }
        public RoundDirection Alle_Arrotondamento_Verso { get; set; }
        public Boolean Alle_Use_4_Match { get; set; }

        public int NumCoppia { get; set; } = 0;
        public int IdCausale_HH_Lav { get; set; } = 0;

    }

    public class Par_OrarioIntervalloHHInModel
    {

    }

    public class Par_OrarioIntervalloHHOutModel : ModelResult
    {
        public List<Par_OrarioIntervalloHHModel> Par_OrarioIntervalloHH { get; set; } = new List<Par_OrarioIntervalloHHModel>();

        public Par_OrarioIntervalloHHOutModel()
        {

        }
    }



    public class Par_OrarioIntervalloHH_Get_4Edit_InModel
    {
        public int Id { get; set; }  // id del orario
    }
    public class Par_OrarioIntervalloHH_Get_4Edit_OutModel
    {
        public List<Par_OrarioIntervalloHHModel> Par_OrarioIntervalloHH { get; set; } = new List<Par_OrarioIntervalloHHModel>();
    }

    public class Par_OrarioIntervalloHH_Put_4Edit_InModel
    {
        public int Id { get; set; }  // id del orario
        public List<Par_OrarioIntervalloHHModel> Par_OrarioIntervalloHH { get; set; } = new List<Par_OrarioIntervalloHHModel>();
    }
    public class Par_OrarioIntervalloHH_Put_4Edit_OutModel
    {
        public int Id { get; set; }  // id del orario
        public List<Par_OrarioIntervalloHHModel> Par_OrarioIntervalloHH { get; set; } = new List<Par_OrarioIntervalloHHModel>();
    }


    public class Par_OrarioIntervalloHH_Arrange_Coppie_InModel
    {
        public int Id { get; set; }  // id del orario
        public int NumCoppie { get; set; }  
        public List<Par_OrarioIntervalloHHModel> Par_OrarioIntervalloHH { get; set; } = new List<Par_OrarioIntervalloHHModel>();
    }
    public class Par_OrarioIntervalloHH_Arrange_Coppie_OutModel
    {
        public int Id { get; set; }  // id del orario
        public List<Par_OrarioIntervalloHHModel> Par_OrarioIntervalloHH { get; set; } = new List<Par_OrarioIntervalloHHModel>();
    }

}

File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Par_OrarioIntervalloHH/Models/par-orario-intervallo-hh-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";



export class Par_OrarioIntervalloHHModel {
  id: number;
  idPar_Orario: number;

  public dalle: string | null;
  public dalle_Limite_SX: string | null;
  public dalle_Limite_DX: string | null;
  public dalle_Arrotondamento: TimeRoundInterval;
  public dalle_Arrotondamento_Verso: RoundDirection;

  public alle: string | null;
  public alle_Limite_SX: string | null;
  public alle_Limite_DX: string | null;
  public alle_Arrotondamento: TimeRoundInterval;
  public alle_Arrotondamento_Verso: RoundDirection;

  numCoppia: number;
}


export enum TimeRoundInterval {
  Min1 = 1,
  Min5 = 5,
  Min15 = 15,
  Min30 = 30,
  Min60 = 60
}

export enum RoundDirection {
  Down,
  Up
}



export class Par_OrarioIntervalloHHInModel {
  
}
export class Par_OrarioIntervalloHHOutModel extends ModelResult {

  public par_OrarioIntervalloHH: Par_OrarioIntervalloHHModel[] = [];

}



// Modelli per la gestione dell'edit
export class Par_OrarioIntervalloHH_Get_4Edit_InModel {
  id: number = 0; // id del profilo
}
export class Par_OrarioIntervalloHH_Get_4Edit_OutModel extends ModelResult {
  par_OrarioIntervalloHHM: Par_OrarioIntervalloHHModel[] = [];
}

export class Par_OrarioIntervalloHH_Put_4Edit_InModel {
  id: number = 0; // id del profilo
  par_OrarioIntervalloHHM: Par_OrarioIntervalloHHModel[] = [];
}
export class Par_OrarioIntervalloHH_Put_4Edit_OutModel extends ModelResult {
  id: number = 0; // id del profilo
  par_OrarioIntervalloHHM: Par_OrarioIntervalloHHModel[] = [];
}


export class Par_OrarioIntervalloHH_Arrange_Coppie_InModel {
  id: number = 0; // id del profilo
  numCoppie: number = 0; 
  par_OrarioIntervalloHH: Par_OrarioIntervalloHHModel[] = [];
}
export class Par_OrarioIntervalloHH_Arrange_Coppie_OutModel extends ModelResult {
  id: number = 0; // id del profilo
  par_OrarioIntervalloHH: Par_OrarioIntervalloHHModel[] = [];
}
